parser.go

```diff
@@ -315,6 +315,16 @@ func (p *parsing) parseNumber() bool {
 				} else {
 					break
 				}
+			} else if !hasExp && p.curr == '-' { // Allow starting numbers from a negative sign.
```

Comment:
I analyzed this and realized it breaks backward compatibility. Let's make it a separate setting, something like "negative number parsing" settings.